server/src/main/java/io/crate/types/ArrayType.java

```diff
@@ -89,12 +92,17 @@ public ArrayType(DataType<T> innerType) {
         StorageSupport innerStorage = innerType.storageSupport();
         if (innerStorage == null) {
             this.storageSupport = null;
+        } else if (ArrayType.unnest(this) instanceof ObjectType || ArrayType.unnest(this) instanceof GeoShapeType) {
+            this.storageSupport = new StorageSupport<T>(innerStorage) {
```

Comment:
This could also use a comment. Especially the GeoShapeType case.

Although it is currently using a Map for value representation it is not a container type like ObjectType.